src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -454,28 +511,46 @@ private async Task ProcessIncomingAsync(ConnectionContext connection)
                 }
             }
             finally
-            {
-                keepAliveTimer.Stop();
-            }
+            {
+                _closeIncomingLock.Release();
+                keepAliveTimer.Stop();
+                Status = ServiceConnectionStatus.Disconnected;
+                await DisposeConnection(connection);
+            }
+
+            // TODO: Never cleanup connections unless Service asks us to do that
+            // Current implementation is based on assumption that Service will drop clients
+            // if server connection fails.
+            await CleanupConnections();
         }
 
         private Task DispatchMessageAsync(ServiceMessage message)
         {
-            switch (message)
-            {
-                case OpenConnectionMessage openConnectionMessage:
-                    return OnConnectedAsync(openConnectionMessage);
-                case CloseConnectionMessage closeConnectionMessage:
-                    return OnDisconnectedAsync(closeConnectionMessage);
-                case ConnectionDataMessage connectionDataMessage:
-                    return OnMessageAsync(connectionDataMessage);
-                case ServiceErrorMessage serviceErrorMessage:
-                    return OnServiceErrorAsync(serviceErrorMessage);
-                case PingMessage pingMessage:
-                    return OnPingMessageAsync(pingMessage);
-                case AckMessage ackMessage:
-                    return OnAckMessageAsync(ackMessage);
+            _clientConnectionLock.WaitAsync();
```

Comment:
looks useless.

Reply:
You're right, it was once useful and can be dropped now.